md-dao/src/main/java/com/wipro/ats/bdre/md/dao/ProcessAncestorsDAO.java

```diff
@@ -47,19 +49,19 @@
 
         try {
             session.beginTransaction();
-            Criteria checkParentProcess = session.createCriteria(com.wipro.ats.bdre.md.dao.jpa.Process.class).add(Restrictions.eq("processId", processId)).add(Restrictions.isNull("process.processId"));
+            Criteria checkParentProcess = session.createCriteria(com.wipro.ats.bdre.md.dao.jpa.Process.class).add(Restrictions.eq(PROCESSID, processId)).add(Restrictions.isNull(PARENTPROCESSID));
             Integer parentProcessCount = checkParentProcess.list().size();
             LOGGER.info("Parent process count:" + parentProcessCount);
             if (parentProcessCount == 0) {
                 throw new MetadataException("Invalid parent process:" + processId);
             }
 
-            Criteria checkEnqueuingProcesses = session.createCriteria(com.wipro.ats.bdre.md.dao.jpa.Process.class).add(Restrictions.eq("process.processId", processId)).add(Restrictions.eq("deleteFlag", false)).setProjection(Projections.distinct(Projections.property("enqueuingProcessId")));
+            Criteria checkEnqueuingProcesses = session.createCriteria(com.wipro.ats.bdre.md.dao.jpa.Process.class).add(Restrictions.eq(PARENTPROCESSID, processId)).add(Restrictions.eq("deleteFlag", false)).setProjection(Projections.distinct(Projections.property("enqueuingProcessId")));
             List<Integer> enqueuingProcessIdList = checkEnqueuingProcesses.list();
 
             LOGGER.info("Number of enqueuing process count:" + checkEnqueuingProcesses.list().size());
             if (enqueuingProcessIdList.size() != 0) {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Use isEmpty() to check whether the collection is empty or not. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonar.openbdre.org/coding_rules#rule_key=squid%3AS1155)